File: 1352-product-of-the-last-k-numbers/1352-product-of-the-last-k-numbers.py
class ProductOfNumbers(object):
    listProd=[]
    def __init__(self):        
        self.listProd.append(1)

    def add(self, num):
        """
        :type num: int
        :rtype: None
        """
        if num==0:
            self.listProd=[]  
            self.listProd.append(1)
        else:
            self.listProd.append(self.listProd[-1]*num)
        

    def getProduct(self, k):
        """
        :type k: int
        :rtype: int
        """
        if k>len(self.listProd)-1:
            return 0
        else:
            return self.listProd[-1]/self.listProd[-1-k]
        


# Your ProductOfNumbers object will be instantiated and called as such:
# obj = ProductOfNumbers()
# obj.add(num)
# param_2 = obj.getProduct(k)

File: 1352-product-of-the-last-k-numbers/1352-product-of-the-last-k-numbers.cpp
int product[40000];
class ProductOfNumbers {
public:
    int n=1;
    ProductOfNumbers(){
        product[0]=1;
    }
    
    void add(int num) {
        if (num==0) {
            n=1;
        }
        else{
            product[n]=product[n-1]*num;
            n++;
        }
    }
    
    int getProduct(int k) {
        if (n<=k) return 0;
        return product[n-1]/product[n-k-1];
    }
};

/**
 * Your ProductOfNumbers object will be instantiated and called as such:
 * ProductOfNumbers* obj = new ProductOfNumbers();
 * obj->add(num);
 * int param_2 = obj->getProduct(k);
 */